### PR TITLE
upgrade node engine

### PR DESCRIPTION
**Pull request by ahmetcetin**
### Description
Node v10 is in LTS now. Tested with node 10.14.2, and it works.

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by ahmetcetin** [comment]
closes #41

File: package.json
Comment (dhmlau) [inline package.json]: there's no need to change the version.  When we release, it will update the version number automatically.

File: README.md
Comment (dhmlau) [inline README.md:247]: I believe there's no change in this line.

File: package.json
Comment (dhmlau) [inline package.json:6]: I'm thinking whether we should just make it `>=6`. 
@strongloop/loopback-maintainers, do you know any particular reasons why we want to specify a particular node.js version? Thanks.

**Review by dhmlau — COMMENTED** [review]
@ahmetcetin , thanks for your PR.  Please address my comments in this PR.
Also, I have a question to the maintainers on whether we can just set the `engines` to be `>=6` instead of specifying particular versions of Node.js.

File: README.md
Comment (ahmetcetin) [reply README.md:247]: well, i didn't touch this line, probably white space thing.

File: package.json
Comment (ahmetcetin) [reply package.json:6]: i first did like that, but later changed it to keep the same way it was kept before. cls-hooked is using async_hooks api of node if node version is >=8.2.1. it's still in experimental status, so i think it's better to keep the node version under control instead of >6.0. there might be breaking changes still on this api until it becomes stable, that's my 2c.

File: package.json
Comment (ahmetcetin) [reply package.json]: ok, changed back the version.

**Review by dhmlau — APPROVED** [review]
Would like to get one more approval before landing.
@jannyHou @b-admike, could you PTAL?  Thanks.

**Comment by jannyHou** [comment]
The change makes sense to me and I am good with using a union to specify the node version for experimental features.

@ahmetcetin Thank you for the contribution! A question: how do you find that particular Node.js 10 version that supports this feature? I couldn't find it in https://node.green/, or you just point it to the latest minor version which is 10.14?

**Comment by ahmetcetin** [comment]
in node's documentation: https://nodejs.org/api/async_hooks.html, that's what cls-hooked is using for the node versions above 8.2

**Review by jannyHou — APPROVED** [review]
@ahmetcetin Cool. Could you squash the 6 commits to one, I will help you land the PR.

**Comment by ahmetcetin** [comment]
any easy way to do it, or should i create another PR?

**Comment by dhmlau** [comment]
@ahmetcetin , about squashing commits, you can refer to this article: https://github.com/wprig/wprig/wiki/How-to-squash-commits.

**Comment by ahmetcetin** [comment]
closing this PR in favor of PR #43

**Pull request closed without merging**
